src/transaction_builder.cpp

```diff
@@ -141,6 +143,16 @@ void TransactionBuilder::AddSaplingSpend(
         throw std::runtime_error("TransactionBuilder cannot add Sapling spend to pre-Sapling transaction");
     }
 
+    unsigned char leadbyte = 0x01;
+    if (note.get_is_zip_212() == true) {
+        leadbyte = 0x02;
+    }
+
+    // ZIP212: check that note plaintext lead byte is valid at height
+    if (!libzcash::plaintext_version_is_valid(consensusParams, nHeight + 1, leadbyte)) {
```

Comment:
I believe we don't want to do this, right? In this context, we're adding a _spend_ to the transaction builder. That means we're spending a note that we already "received" and therefore we know the plaintext version is correct on chain. Furthermore, this check would check what we're spending against the next block's height. I believe that would cause this exception to be thrown if we're spending a note that was created before canopy activated.

Reply:
yes, in fact we had a test for exactly that! getting rid of this check in `AddSaplingSpend` + the test in `test_transaction_builder.cpp`